src/votecoding.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Candidate,
+  CandidateContest,
+  CandidateVote,
+  Contests,
+  Vote,
+  VotesDict,
+} from './config/types'
+
+export default function encodeVotes(contests: Contests, votes: VotesDict) {
+  return contests
+    .map(contest => {
+      const contestVote: Vote = votes[contest.id]!
+      if (!contestVote) {
+        return ''
+      }
+
+      if (contest.type === 'yesno') {
+        if (contestVote === 'yes') {
+          return '1'
+        } else {
+          return '0'
+        }
+      }
+
+      const candidateIDs = (contest as CandidateContest).candidates.map(
+        (c: Candidate) => c.id
+      )
+      return (contestVote as CandidateVote)
+        .map(c => (c.isWriteIn ? 'W' : candidateIDs.indexOf(c.id)))
```

Comment:
Why not encode the write-in value submitted by the voter such that the ballot QR Code and the write-in value printed on the ballot?

Reply:
because
1/ space is limited in QR code, and a few write-ins will shrink the pixels and make it harder to decode.
2/ then we have to worry about escaping characters properly so our parsing doesn't get in trouble (super tricky to do fully well without going full JSON)

Reply:
also, write-ins tend to be a different process, not counted unless they go above a certain amount.